components/timezone/src/metazone.rs

```diff
@@ -19,38 +19,46 @@ pub struct MetazoneCalculator {
     pub(super) metazone_period: DataPayload<MetazonePeriodV1Marker>,
 }
 
+#[cfg(feature = "data")]
+impl Default for MetazoneCalculator {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
 impl MetazoneCalculator {
-    /// Constructor that loads data before calculating metazone id.
-    ///
-    /// [📚 Help choosing a constructor](icu_provider::constructors)
-    /// <div class="stab unstable">
-    /// ⚠️ The bounds on this function may change over time, including in SemVer minor releases.
-    /// </div>
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// use icu::timezone::MetazoneCalculator;
-    /// use icu_locid::locale;
+    /// Constructs a `MetazoneCalculator`.
     ///
-    /// let mzc = MetazoneCalculator::try_new_unstable(&icu_testdata::unstable());
+    /// ✨ **Enabled with the `"data"` feature.**
     ///
-    /// assert!(mzc.is_ok());
-    /// ```
-    pub fn try_new_unstable<P>(zone_provider: &P) -> Result<Self, TimeZoneError>
-    where
-        P: DataProvider<MetazonePeriodV1Marker> + ?Sized,
-    {
-        let metazone_period = zone_provider
-            .load(DataRequest {
-                locale: Default::default(),
-                metadata: Default::default(),
-            })?
-            .take_payload()?;
-        Ok(Self { metazone_period })
+    /// [📚 Help choosing a constructor](icu_provider::constructors)
+    #[cfg(feature = "data")]
+    pub fn new() -> Self {
```

Comment:
Thought: maybe in cases where we could have a `pub fn new()`, we should instead export a static instance, which is probably better than people inlining calls to `new()` everywhere.

Reply:
I mention this in the pr description. I think for constructors `new` is more consistent with the provider constructors, and gets around the issue of naming the singleton. Performance is probably better with a const instead of a static, as that will be inlined.

For things like `load_emoji` I'm planning to have an `EMOJI` const.